compiler/ilgen/OMRIlBuilder.cpp

```diff
@@ -2679,6 +2679,20 @@ OMR::IlBuilder::MakeCase(int32_t caseValue, TR::IlBuilder **caseBuilder, int32_t
    return c;
    }
 
+TR::IlValue *
+OMR::IlBuilder::Select(TR::IlValue * condition, TR::IlValue * trueValue, TR::IlValue * falseValue){
+   TR_ASSERT(condition != NULL && trueValue != NULL && falseValue != NULL,
+                     "SelectIf requires condition, trueValue and falseValue");
+   TR_ASSERT(trueValue->getDataType() == falseValue->getDataType(), "Value type mismatch");
+   TR::Node * conditionNode = loadValue(condition);
+   TR::Node * ifTrueNode = loadValue(trueValue);
+   TR::Node * ifFalseNode = loadValue(falseValue);
+   TR::ILOpCodes opCode = TR::ILOpCode::ternaryOpCode(trueValue->getDataType());
+   TR::Node * resultNode = createWithoutSymRef(opCode, 3, conditionNode, ifTrueNode, ifFalseNode);
```

Comment:
with more tolerant API from TR::ILOpCode, you could decide here to implement the `Select` with an `IfThenElse`...

Reply:
Not sure how I could implement `Select` with `IfThenElse` here..

Reply:
If you're talking about creating a new temp and builders that store over the new temp depending on the condition, `Select` will be able to handle the unimplemented data types (ie, floats and doubles on x86). I've been planning on implementing `fternary` and `dternary` evaluators for x86 in a follow-up PR soon, so is this workaround using `IfThenElse` still necessary?

Reply:
Another option is to have `TR::IlOpCode::ternaryOpCode` return `BadIlOpCode` for floats, doubles and vectors, and use the `IfThenElse` way in that case. I think this is the best option to go for.